### PR TITLE
feat(store-engine): Added Store Engine

### PR DESCRIPTION
**Pull request by bennyn**

### CODE REVIEW

File: packages/store-engine/README.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+Nowadays there are more and more storage possibilities and developers must be familiar with the characteristics of each individual solution to reliably store data. Because it can be sometimes hard to keep up with the highly dynamic world of data storages, we have developed a system which unifies the usage of [IndexedDB](https://developer.mozilla.org/docs/IndexedDB), [In-memory storage](https://en.wikipedia.org/wiki/In-memory_database), [File-based storage](https://nodejs.org/api/fs.html) and [LocalStorage](https://developer.mozilla.org/docs/Web/API/Window/localStorage). In addition we built some functionality (like a transient store which deletes data after a [TTL](https://en.wikipedia.org/wiki/Time_to_live)) on top.
+
+### Overview
+
+|       Engine       | Available in Browser | Available in Node.js |                                                        Description                                                         |
+| :----------------: | :------------------: | :------------------: | :------------------------------------------------------------------------------------------------------------------------: |
```
Comment (lipis) [inline packages/store-engine/README.md]: I think the checkboxes is OK to have them centered.. but not the rest.

File: packages/store-engine/README.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+|     FileEngine     |          ❌          |          ✔           |                        Rudimentary persistent store based on files. Very generic and easy to read.                         |
+|  IndexedDBEngine   |          ✔           |          ❌          | Persisent storage which handles significant amounts of structured data, including files/blobs. Enables very fast searches. |
+|    MemoryEngine    |          ✔           |          ✔           |                Transient store which loses data on application restart. Suitable for testing environments.                 |
+| LocalStorageEngine |          ✔           |          ❌          | Can save very small amount of data. Stored data is saved across browser sessions. Suitable for simple objects and strings. |
+
+### Quickstart
+
+#### Engines
+
+**Hooking up a FileEngine**
```
Comment (lipis) [inline packages/store-engine/README.md]:
```md
##### Hooking up a FileEngine
```

File: packages/store-engine/README.md (hunk 1 above)
Comment (bennyn) [reply packages/store-engine/README.md]: Prettier messed up with my whole table:

unbenannt

It was much easier to edit without it:

unbenannt

Comment (lipis) [reply packages/store-engine/README.md]: Then just add `<!-- prettier-ignore -->` in front of it..  and you are done :)

Comment (bennyn) [reply packages/store-engine/README.md]: 👍

**Pull request merged**